api/python/provisioner/commands/check.py

```diff
@@ -0,0 +1,392 @@
+                                fun_kwargs=dict(python_shell=True))
+            # TODO: parse command output if necessary
+            res[minion_id] = f"{cfg.CheckVerdict.PASSED}: {str(_res)}"
+        except SaltCmdResultError as e:
+            res[minion_id] = f"{cfg.CheckVerdict.FAIL.value}: {str(e)}"
+
+        return res
+
+    def _connectivity(self, *, args: str,
+                      servers: Union[list, tuple, set] = PENDING_SERVERS,
+                      dry_run: bool = False) -> dict:
+        """
+        Check servers connectivity. Ping servers to check for availability
+
+        :param str args: possible arguments for connectivity checks
+        :param Union[list, tuple, set] servers: servers for pinging
+        :param dry_run: Execute method without real check execution on
+                        target nodes
+        :return:
+        """
+        if dry_run:
+            return dict()
+
+        res = dict()
+        for addr in servers:
+            # NOTE: check ping of 'srvnode-2' from 'srvnode-1'
+            # and vise versa
+            # TODO: which targets do we need to use? Because we need to
+            #  check cross connectivity if nodes > 2
+            targets = set(servers) - {addr}
+            targets = (cfg.LOCAL_MINION if not targets
+                       else next(iter(targets)))  # takes just one node
+
+            cmd = self._PING_CMD_TMP.format(server_addr=addr)
+            try:
+                function_run("cmd.run", targets=targets, fun_args=[cmd])
+            except SaltCmdResultError:
+                res[targets] = (f"{cfg.CheckVerdict.FAIL.value}: "
+                                f"{addr} is not reachable from {targets}")
+            else:
+                res[targets] = cfg.CheckVerdict.PASSED.value
+
+    def _logs_are_good(self, *, args: str, targets: str = cfg.ALL_MINIONS,
+                       dry_run: bool = False) -> dict:
+        """
+        Check that logs are clear and don't contain some specific key phrases
+        which signal about serious issues
+
+        :param args: logs_are_good check specific parameters and arguments
+        :param targets: target nodes where network checks will be executed
+        :param dry_run: Execute method without real check execution on
+                        target nodes
+        :return:
+        """
+        def check_log_file(log_file, key_phrases, _targets):
+            """
+            Check separate log_file against key_phrases
+
+            :param log_file:
+            :param key_phrases:
+            :param _targets
+            :return:
+            """
+            for phrase in key_phrases:
+                cmd = self._LOGS_ARE_GOOD_CHECK_CMD_TMP.format(
+                    log_filename=log_file,
+                    key_phrase=phrase)
+                try:
+                    function_run("cmd.run", targets=_targets, fun_args=[cmd])
+                    res[minion_id] = {
```

Comment:
Are you sure these outer scope vars would work? In any case looks awkward.

Reply:
Yes, moved vars definition before this function